src/klr.net45/klr.net45.cpp

```diff
@@ -16,13 +16,24 @@ extern "C" __declspec(dllexport) HRESULT __stdcall CallApplicationMain(PCALL_APP
     IKatanaManagerPtr manager = new ComObject<KatanaManager, IKatanaManager>();
 
     g_katanaManager = manager;
-    _HR(manager->InitializeRuntime(data->applicationBase));
-    g_katanaManager = NULL;
 
-    _HR(manager->CallApplicationMain(data->argc, data->argv));
+    hr = manager->InitializeRuntime(data->applicationBase);
+    if (FAILED(hr))
+    {
+        printf_s("Failed to initialize runtime (%d).\n", hr);
+        data->exitcode = hr;
 
-    data->exitcode = hr;
+        return hr;
+    }
 
+    g_katanaManager = NULL;
+    hr = manager->CallApplicationMain(data->argc, data->argv);
+    if (FAILED(hr))
+    {
+        printf_s("Application returns result indicating failure.");
```

Comment:
@davidfowl @Eilon @muratg  We need some guideline in terms of how much information we disclose in the error message.